test/images/sample-device-plugin/sampledeviceplugin.go

```diff
@@ -84,14 +92,59 @@ func main() {
 		klog.Errorf("Empty pluginSocksDir")
 		return
 	}
+
+	autoRegisterString := os.Getenv("AUTO_REGISTER")
+	if autoRegisterString == "" {
+		autoRegister = true
+	} else {
+
+		autoRegister, err = strconv.ParseBool(autoRegisterString)
+		if err != nil {
+			klog.Errorf("invalid boolean value specified")
+			panic(err)
+		}
+
+	}
+
 	socketPath := pluginSocksDir + "/dp." + fmt.Sprintf("%d", time.Now().Unix())
+	triggerPath := pluginSocksDir + "/registered"
 
 	dp1 := plugin.NewDevicePluginStub(devs, socketPath, resourceName, false, false)
 	if err := dp1.Start(); err != nil {
 		panic(err)
 
 	}
 	dp1.SetAllocFunc(stubAllocFunc)
+
+	if !autoRegister {
+		l, err := net.Listen("unix", triggerPath)
+		klog.Infof("Started Listening at %s", triggerPath)
+		if err != nil {
+			panic(err)
+		}
+		defer l.Close()
+
+		sigc := make(chan os.Signal, 1)
+		signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
+		go func(ln net.Listener, c chan os.Signal) {
+			sig := <-c
+			klog.Infof("Received termination signal %s: shutting down.", sig)
+			ln.Close()
+			os.Exit(0)
+		}(l, sigc)
+
+		klog.InfoS("Starting to accept connections")
+		// Accept new connections
+		conn, err := l.Accept()
+		if err != nil {
+			panic(err)
+		}
+
+		defer conn.Close()
+		io.Copy(conn, conn)
```

Comment:
why is this needed?

Reply:
I recall playing around with handling connections in separate goroutines but later made changes and this might be a reminiscence of that. Will remove it.